src/Sulu/Bundle/MediaBundle/Resources/public/js/main.js

```diff
@@ -56,28 +57,45 @@ define([
                 };
             });
 
-            // list all collections
+            // list the top collections with all media
             sandbox.mvc.routes.push({
-                route: 'media/collections/root',
+                route: 'media/collections/:locale',
+                callback: function(locale) {
+                    return '<div data-aura-component="collections/edit@sulumedia" data-aura-locale="' + locale + '"/>';
+                }
+            });
+
+            // list the top collections with all media (without locale)
+            sandbox.mvc.routes.push({
+                route: 'media/collections',
                 callback: function() {
-                    return '<div data-aura-component="collections/root@sulumedia"/>';
+                    MediaRouter.toRootCollection(UserSettingsManager.getMediaLocale());
                 }
             });
 
             // show a single collection with files and upload
             sandbox.mvc.routes.push({
-                route: 'media/collections/edit::id/:content',
-                callback: function(id) {
-                    return '<div data-aura-component="collections/edit@sulumedia" data-aura-id="' + id + '"/>';
+                route: 'media/collections/:locale/edit::id/:content',
+                callback: function(locale, id) {
+                    return '<div data-aura-component="collections/edit@sulumedia" data-aura-id="' + id + '" data-aura-locale="' + locale + '"/>';
                 }
             });
 
             // show a single collection with files and upload
             sandbox.mvc.routes.push({
+                route: 'media/collections/:locale/edit::id/:content/edit::mediaId',
+                callback: function(locale, id, content, mediaId) {
+                    sandbox.sulu.viewStates['media-file-edit-id'] = parseInt(mediaId);
```

Comment:
From a pure feeling this is not really correct here... Shouldn't this be handled by the component somehow? This routing file is supposed to be as simple as possible.